File: src/main/java/com/gmail/borlandlp/minigamesdtools/config/ConfigProvider.java
package com.gmail.borlandlp.minigamesdtools.config;

import com.gmail.borlandlp.minigamesdtools.config.exception.DuplicateConfigException;

import java.util.List;

public interface ConfigProvider {
    void registerEntity(ConfigPath p, ConfigEntity file) throws DuplicateConfigException;
    void registerEntity(String poolId, ConfigEntity file) throws DuplicateConfigException;
    ConfigEntity getEntity(ConfigPath p, String entityId);
    ConfigEntity getEntity(String poolId, String entityId);
    List<ConfigEntity> getPoolContents(ConfigPath path);
    List<ConfigEntity> getPoolContents(String poolId);
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/config/ConfigProvider.kt
package com.gmail.borlandlp.minigamesdtools.config

import com.gmail.borlandlp.minigamesdtools.config.exception.DuplicateConfigException

interface ConfigProvider {
    @Throws(DuplicateConfigException::class)
    fun registerEntity(p: ConfigPath, file: ConfigEntity)

    @Throws(DuplicateConfigException::class)
    fun registerEntity(poolId: String, file: ConfigEntity)

    fun getEntity(p: ConfigPath, entityId: String): ConfigEntity?
    fun getEntity(poolId: String, entityId: String): ConfigEntity?
    fun getPoolContents(path: ConfigPath): List<ConfigEntity>
    fun getPoolContents(poolId: String): List<ConfigEntity>
}